sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { ZBClient } from "zeebe-node";
+import { Job } from "bull";
+import { createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { CancelAssessmentQueueInDTO } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../test/helpers";
+import { CancelApplicationAssessmentProcessor } from "../cancel-application-assessment.processor";
+import { DataSource, Repository } from "typeorm";
+import {
+  createFakeDisbursementOveraward,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  ApplicationStatus,
+  DisbursementOveraward,
+  DisbursementSchedule,
+  DisbursementScheduleStatus,
+  StudentAssessment,
+} from "@sims/sims-db";
+import * as faker from "faker";
+
+describe(
+  describeProcessorRootTest(QueueNames.CancelApplicationAssessment),
+  () => {
+    let app: INestApplication;
+    let processor: CancelApplicationAssessmentProcessor;
+    let zbClientMock: ZBClient;
+    let appDataSource: DataSource;
+    let studentAssessmentRepo: Repository<StudentAssessment>;
+    let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+    let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, zbClient } =
+        await createTestingAppModule();
+      app = nestApplication;
+      zbClientMock = zbClient;
+      appDataSource = dataSource;
+      studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+      disbursementOverawardRepo = dataSource.getRepository(
+        DisbursementOveraward,
+      );
+      disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+      // Processor under test.
+      processor = app.get(CancelApplicationAssessmentProcessor);
+    });
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it("Should cancel the assessment pending disbursements and rollback overawards when the cancelled application has overawards and also one sent and one pending disbursements.", async () => {
+      // Arrange
+      const workflowInstanceId = faker.random.number({
+        min: 1000000000,
+        max: 9999999999,
+      });
```

Comment:
we can put the .toString() here to have the workflowInstanceId as string to use in other places.

```ts

 const workflowInstanceId = faker.random.number({
        min: 1000000000,
        max: 9999999999,
      }).toString();
```

Reply:
Changed.